### PR TITLE
Add support for automatically compute unit test coverage

### PR DESCRIPTION
**Pull request by rionda**
Add make target "coverage" and supporting code to compute the coverage of unit tests.

### CODE REVIEW

**Comment by rionda** [comment]
@zifeishan , if you have time can you please look at this?
Thank you.

File: test.sh [hunk 1]
```diff
@@ -1,20 +1,19 @@
 set -e
 
+# if greenplum supported (gpfdist exist), then test GP
+if hash gpfdist 2>/dev/null; then
+  echo "Testing Greenplum..."
+  bash test/test_gp.sh $1
 # if command psql exist, then test psql
-if hash psql 2>/dev/null; then
+elif hash psql 2>/dev/null; then
```
Comment (zifeishan) [inline test.sh]: @rionda It was intended to test Greenplum twice, one with "test_gp.sh" which tests with parallel grounding, the other uses non-parallel grounding. Maybe we could find a better structure for this?

Comment (rionda) [inline test.sh]: @zifeishan : I see. I didn't get that. Ok, I'm going to revert it back to the original, but I'll add a comment explaining what you told me.

**Comment by zifeishan** [comment]
Looks good to me. @rionda

**Pull request merged**